gammapy/maps/hpx.py

```diff
@@ -1732,6 +1732,28 @@ def __repr__(self):
         str_ += "\tcenter     : {lon:.1f} deg, {lat:.1f} deg\n".format(lon=lon, lat=lat)
         return str_
 
+    def _check_compatibility(self, other):
+        # check overall shape and axes compatibility
+        if self.data_shape != other.data_shape:
+            raise ValueError("MapGeom data shapes differ")
+        for axis, otheraxis in zip(self.axes, other.axes):
+            axis._check_compatibility(otheraxis)
+
+        # check healpix geometry consistency
+        if self.nside != other.nside:
+            raise ValueError("HpxGeom differ")
+        if self.coordsys != other.coordsys:
+            raise ValueError("HpxGeom differ")
+        if self.order != other.order:
+            raise ValueError("HpxGeom differ")
+        if self.center_pix != other.center_pix:
+            raise ValueError("HpxGeom differ")
+        if self.center_coord != other.center_coord:
```

Comment:
Do we want exact float comparison here?

This usually works for in-memory copies, but often fails after I/O, because parameters are either stored in 32 bit floats or via string representations in FITS headers.

My initial feeling is that we don't need bit-by-bit equality checks, and could just always do allclose on float, and that would cover our use cases?

Reply:
center_pix and center_coord are not needed for all sky healpix maps if I understand correctly anyway. They are removed.